src/Analysis/Ast/Impl/Extensions/NodeExtensions.cs

```diff
@@ -54,5 +54,12 @@ public static class NodeExtensions {
             }
             return LocationInfo.Empty;
         }
+
+        public static bool IsInAst(this ScopeStatement node, PythonAst ast) {
```

Comment:
Should this return `true` if `node` is `ast` itself?

Reply:
Interesting. Probably yes.

Reply:
In fact it is does `ast.Parent` is null so loop is bypassed and `ast == node` will do the right thing.